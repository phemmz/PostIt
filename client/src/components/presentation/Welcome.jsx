import React from 'react';

const Welcome = () => {
  return (
    <div>
      <div className="container row">
        <div className="col s7 offset-s3 welc">
          <h4 className="green-text text-darken-4">Welcome,</h4>
          <p
            className="green-text text-darken-2"
          >POSTIT allows you, your friends and
          colleagues to create groups for notifications.
          </p>
        </div>
      </div>
      <footer className="black footer">
        <div className="ffooter footer-copyright">
          <div className="container">
        &copy; <script
          type="text/javascript"
        >document.write(new Date().getFullYear());</script> Adetunji Femi
          </div>
        </div>
      </footer>
    </div>
  );
};

export default Welcome;
