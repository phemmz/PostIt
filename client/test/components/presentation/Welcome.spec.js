import React from "react";
import { mount } from "enzyme";
import Welcome from '../../../src/components/presentation/Welcome';

describe("Welcome Component", () => {
  let mountedWelcome;

  const welcome = () => {
    if (!mountedWelcome) {
      mountedWelcome = mount(
        <Welcome />
      );
    }
    return mountedWelcome;
  }

  beforeEach(() => {
    mountedWelcome: undefined;
  });

  it("should always render a div", () => {
    const divs = welcome().find("div");
    expect(welcome()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  it("should contain everything else that gets rendered", () => {
    const divs = welcome().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(welcome().children());
  });
});
