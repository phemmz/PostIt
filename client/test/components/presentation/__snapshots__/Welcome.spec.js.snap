// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component should always render a div 1`] = `
<Welcome>
  <div>
    <div
      className="container row"
    >
      <div
        className="col s7 offset-s3 welc"
      >
        <h4
          className="green-text text-darken-4"
        >
          Welcome,
        </h4>
        <p
          className="green-text text-darken-2"
        >
          POSTIT allows you, your friends and colleagues to create groups for notifications.
        </p>
      </div>
    </div>
    <footer
      className="black footer"
    >
      <div
        className="ffooter footer-copyright"
      >
        <div
          className="container"
        >
          Â© 
          <script
            type="text/javascript"
          >
            document.write(new Date().getFullYear());
          </script>
           Adetunji Femi
        </div>
      </div>
    </footer>
  </div>
</Welcome>
`;
